syntax = "proto3";

package dylk.patio;

// -------------------------------------------------------------------------
// The PatioVouchers service definition.
service PatioVouchers {
  rpc AlterPatioVoucher   (PatioVoucher)            returns (PatioVoucherGeneralResponse)  {}
  rpc ListPatioVouchers   (PatioVoucherParams)      returns (PatioVoucherListResponse)     {}
  rpc GetPatioVoucher     (PatioVoucherId)          returns (PatioVoucherResponse)         {}
}

message Item {
  string equipmentCode = 1;
  int32 quantity       = 2;
}

message PatioVoucher {
  int32  id                  = 1;
  int32  voucherId           = 2;
  string patioCode           = 3;
  string observations        = 4;
  string deliveredBy         = 5;
  string receivedBy          = 6;
  string status              = 7;
  repeated Item itemList     = 8;
  double generationTime      = 9;
  double lastTouchTime       = 10;
}

message PatioVoucherId {
  int32 id = 1;
}

message PatioVoucherParam {
  string name  = 1;
  string value = 2;
}

message PatioVoucherParams {
  repeated PatioVoucherParam paramList     = 1;
  repeated PatioVoucherParam pageParamList = 2;
}

message PatioVoucherResponse {
  int32 returnCode            = 1;
  string returnMessage        = 2;
  PatioVoucher patioVoucher   = 3;
}

message PatioVoucherListResponse {
  int32 returnCode                       = 1;
  string returnMessage                   = 2;
  repeated PatioVoucher patioVoucherList = 3;
  int32 totalItems                       = 4;
  int32 totalPages                       = 5;
}

message PatioVoucherGeneralResponse {
  int32 returnCode     = 1;
  string returnMessage = 2;
}
